### Step 1 : Install and import libraries

In [1]:
!pip install pulp
from pulp import LpMaximize, LpProblem, LpVariable, lpSum, value

   ---------------------------------------- 0.0/16.4 MB ? eta -:--:--
   ---------------------------------------- 0.0/16.4 MB ? eta -:--:--
    --------------------------------------- 0.3/16.4 MB ? eta -:--:--
   --- ------------------------------------ 1.3/16.4 MB 2.6 MB/s eta 0:00:06
   ------- -------------------------------- 3.1/16.4 MB 4.7 MB/s eta 0:00:03
   -------------- ------------------------- 5.8/16.4 MB 6.6 MB/s eta 0:00:02
   -------------------- ------------------- 8.4/16.4 MB 7.8 MB/s eta 0:00:02
   ----------------------------- ---------- 12.1/16.4 MB 9.6 MB/s eta 0:00:01
   ----------------------------------- ---- 14.7/16.4 MB 10.1 MB/s eta 0:00:01
   ---------------------------------------  16.3/16.4 MB 10.0 MB/s eta 0:00:01
   ---------------------------------------- 16.4/16.4 MB 9.4 MB/s eta 0:00:00


### Step 2 : Define the optimization model

In [3]:
model = LpProblem("Profit_Maximization_Model", LpMaximize)
print(model)

Profit_Maximization_Model:
MAXIMIZE
None
VARIABLES



### Step 3 : Declare decision variables

In [4]:
x1 = LpVariable("Product_A", lowBound=0, cat='Continuous')
x2 = LpVariable("Product_B", lowBound=0, cat='Continuous')

print(f"x1: {x1}")
print(f"x2: {x2}")


x1: Product_A
x2: Product_B


### Step 4 : Define the objective function

In [5]:
model += 40 * x1 + 30 * x2, "Total_Profit"
print(model)


Profit_Maximization_Model:
MAXIMIZE
40*Product_A + 30*Product_B + 0
VARIABLES
Product_A Continuous
Product_B Continuous



### Step 5 : Add constraints

In [6]:
model += 2 * x1 + 1 * x2 <= 100, "Machine_Hours"
model += 3 * x1 + 2 * x2 <= 150, "Labor_Hours"

print(model)


Profit_Maximization_Model:
MAXIMIZE
40*Product_A + 30*Product_B + 0
SUBJECT TO
Machine_Hours: 2 Product_A + Product_B <= 100

Labor_Hours: 3 Product_A + 2 Product_B <= 150

VARIABLES
Product_A Continuous
Product_B Continuous



### Step 6 : Solve the model

- I solve the linear programming problem and check if the optimal solution is found.

In [8]:
from pulp import LpStatus
model.solve()
print(f"Status: {LpStatus[model.status]}")


Status: Optimal


### Step 7 : Print the results

In [9]:
print(f"Optimal Units of Product A: {x1.varValue}")
print(f"Optimal Units of Product B: {x2.varValue}")
print(f"Maximum Profit: ₹{value(model.objective)}")


Optimal Units of Product A: 0.0
Optimal Units of Product B: 75.0
Maximum Profit: ₹2250.0


### Conclusion
- Produce 0 units of Product A and 75 units of Product B.
- This gives a maximum profit of ₹2250.
- Suggest reassessing the profitability of Product A or checking if resource constraints are limiting its production.
